src/lib/upload/upload-file-csv.js

```diff
@@ -0,0 +1,27 @@
+export default class UploadFileCSV {
+  static upload (fileDataArr, delimiter = '\t') {
+    const textDataAll = []
+    let textData = {}
+
+    for (let i = 0; i < fileDataArr.length; i++) {
+      if (fileDataArr[i].indexOf('HEADER:') === 0) {
+        if (i > 0) {
+          textDataAll.push(textData)

```

Comment:
I don't really follow what is happening here. Is this for a file with multiple headers?

Reply:
Yes, it is defined for the case when one file could have several source texts with properties - for uploading a whole alignment (for now only with the source text) via main-menu.
This way I removed converting multiline text to one line (as it was before).
And I believe it would be easier to update such files manually